services/pseudo_controller/src/main/java/com/futurewei/alcor/pseudo_controller/pseudo_controller.java

```diff
@@ -0,0 +1,387 @@
+    static String vpc_id_1 = "2b08a5bc-b718-11ea-b3de-111111111112";
+    static String port_ip_template = "11111111-b718-11ea-b3de-";
+    static String subnet_id_1 = "27330ae4-b718-11ea-b3df-111111111113";
+    static String ips_ports_ip_prefix = "10";
+    static String mac_port_prefix = "6c:dd:ee:";
+    static SortedMap<String, String> ip_mac_map = new TreeMap<>();
+    static Vector<String> aca_node_one_commands = new Vector<>();
+    static Vector<String> aca_node_two_commands = new Vector<>();
+    static SortedMap<String, String> port_ip_to_host_ip_map = new TreeMap<>();
+    static SortedMap<String, String> port_ip_to_id_map = new TreeMap<>();
+    static SortedMap<String, String> port_ip_to_container_name = new TreeMap<>();
+    static Vector<String> node_one_port_ips = new Vector<>();
+    static Vector<String> node_two_port_ips = new Vector<>();
+
+    public static void main(String[] args) {
+        System.out.println("Start of the test controller");
+        if (args.length == 7) {
+            System.out.println("User passed in params and we need to read them.");
+            ports_to_generate_on_each_aca_node = Integer.parseInt(args[0]);
+            aca_node_one_ip = args[1];
+            aca_node_two_ip = args[2];
+            ncm_ip = args[3];
+            ncm_port = Integer.parseInt(args[4]);
+            user_name = args[5];
+            password = args[6];
+
+        }
+        generate_ip_macs(ports_to_generate_on_each_aca_node * 2);
+        create_containers_on_both_hosts();
+        System.out.println("aca_node_one_ip: " + aca_node_one_ip + "\naca_node_two_ip: " + aca_node_two_ip + "\nuser name: " + user_name + "\npassword: " + password);
+
+        System.out.println("Containers setup done, now we gotta construct the GoalStateV2");
+
+        System.out.println("Trying to build the GoalStateV2 for " + ports_to_generate_on_each_aca_node + " Ports");
+
+
+        Goalstate.GoalStateV2.Builder GoalState_builder_one = Goalstate.GoalStateV2.newBuilder();
+        Goalstate.GoalStateV2.Builder GoalState_builder_two = Goalstate.GoalStateV2.newBuilder();
+        Goalstate.HostResources.Builder host_resource_builder_node_one = Goalstate.HostResources.newBuilder();
+        Goalstate.HostResources.Builder host_resource_builder_node_two = Goalstate.HostResources.newBuilder();
+        Goalstate.HostResources.Builder host_resource_builder_node_one_port_one_neighbor = Goalstate.HostResources.newBuilder();
+
+        for (String port_ip : ip_mac_map.keySet()) {
+            String host_ip = port_ip_to_host_ip_map.get(port_ip);
+            String port_id = port_ip_to_id_map.get(port_ip);
+            String port_mac = ip_mac_map.get(port_ip);
+            // if it's on node 1, we don't add neighbor info here,
+            // start of setting up port 1 on aca node 1
+            Port.PortState.Builder new_port_states = Port.PortState.newBuilder();
+
+            new_port_states.setOperationType(Common.OperationType.CREATE);
+
```

Comment:
@zzxgzgz We have changed OperationType for goalstate resources in PM and DPM. I don't know if this change impact the code here. Please refer to PR#602.

Reply:
Thank you for the heads up, will pull the latest change and test again. Modifications will be made if needed.

Reply:
Per our discussion with @xieus, the NCM, at this point, is still using the old contract, thus the TC does not need any change at this moment.